### PR TITLE
networking/tests: add bridge and team tests

### PR DESCRIPTION
**Pull request by HuijingHei**

### CODE REVIEW

**Comment by dustymabe** [comment]
@HuijingHei - I'll try to get to this code review on Monday. Sorry for the delay.

**Comment by HuijingHei** [comment]
> @HuijingHei - I'll try to get to this code review on Monday. Sorry for the delay.

Thanks! No worry, enjoy holiday!

File: tests/kola/networking/bridge-kargs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+##   # Add 2 NIC for this test
+##   additionalNics: 2
+##   # Configuration of the 2 NICs for this test
+##   # We use net.ifnames=0 to disable consistent network naming here because on
+##   # different firmwares (BIOS vs UEFI) the NIC names are different.
+##   # See https://github.com/coreos/fedora-coreos-tracker/issues/1060
+##   appendKernelArgs: "ip=10.10.10.10::10.10.10.1:255.255.255.0:mybridge:br0:none bridge=br0:eth1,eth2 net.ifnames=0"
+##   # appendKernelArgs doesn't work on s390x
+##   # https://github.com/coreos/coreos-assembler/issues/2776
+##   architectures: !s390x
```
Comment (dustymabe) [inline tests/kola/networking/bridge-kargs]: I think this needs quotes. See https://github.com/coreos/fedora-coreos-config/pull/2100

File: tests/kola/networking/team/test.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+##   # https://github.com/coreos/coreos-assembler/issues/2776
+##   architectures: !s390x
+
+set -xeuo pipefail
+
+. $KOLA_EXT_DATA/commonlib.sh
+
+team="team0"
+
+# Verify team0 gets ip address 10.0.2.31 according to config.bu
```
Comment (dustymabe) [inline tests/kola/networking/team/test.sh]: config.bu doesn't specify an IP so this comment at least needs to be updated.

**Review by dustymabe — COMMENTED** [review]
Maybe we should come up with a naming convention for these tests. i.e. something along the lines of:

- `<type>-<static|dhcp>-via-<method>`

So here we would have `bridge-static-via-kargs` and `team-dhcp-via-ignition`.

File: tests/kola/networking/team/test.sh [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+##   platforms: qemu
+##   # Add 2 NIC for this test
+##   additionalNics: 2
+##   # We use net.ifnames=0 to disable consistent network naming here because on
+##   # different firmwares (BIOS vs UEFI) the NIC names are different.
+##   # See https://github.com/coreos/fedora-coreos-tracker/issues/1060
+##   appendKernelArgs: "net.ifnames=0"
+##   # appendKernelArgs doesn't work on s390x
+##   # https://github.com/coreos/coreos-assembler/issues/2776
+##   architectures: !s390x
```
Comment (dustymabe) [inline tests/kola/networking/team/test.sh]: I think this needs quotes. See https://github.com/coreos/fedora-coreos-config/pull/2100

File: tests/kola/networking/team/test.sh (hunk 2 above)
Comment (dustymabe) [reply tests/kola/networking/team/test.sh]: or maybe you meant to test static?

File: tests/kola/networking/team/config.bu [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+          [connection]
+          id=team0
+          type=team
+          autoconnect-retries=1
+          interface-name=team0
+          multi-connect=1
+          [team]
+          config={"runner": {"name": "activebackup"}, "link_watch": {"name": "ethtool"}}
+          [ipv4]
+          dhcp-timeout=90
```
Comment (dustymabe) [inline tests/kola/networking/team/config.bu]:
```suggestion
```

Was this needed?

**Review by dustymabe — APPROVED** [review]
LGTM

**Pull request merged**